app/models/earning.rb

```diff
@@ -1,44 +1,27 @@
-require 'csv'
+# frozen_string_literal: true
+
 class Earning < ApplicationRecord
+  attr_accessor :external_ref
+
   belongs_to :employer
   belongs_to :employee
 
- # before_validations :save_earnings
-
-  def save_earnings(file)
-  	
-  	dates_earning = {}
-  	data = CSV.read(file)
-  	data.each_with_index  do |row, i|
-  		next if i ==0
-  		dates_earning= {external_ref: row[find_employee_id], earning_date: row[find_date_earning], earning_amount: row[find_amout]}
-  	end
-  	employee_info = get_employee(dates_earning[:external_ref])
-  	self.earning_date = format_date(dates_earning[:earning_date])
-  	self.eamount = dates_earning[:earning_amount]
-  end
+  before_validation :find_employee_id
 
-   
-  
-  def self.get_employee(external_ref)
-  	Employee.find_by(external_ref: external_ref)
+  def load_earnings(file)
+    service = EmployerCsvService.new
+    service.employer_id = employer_id
+    service.csv_fields = EmployerCsvLayout.where(employer_id: employer_id)
+    service.process_csv(file)
+  rescue StandardError => e
+    e.message
   end
 
-  def self.find_amout
-  	 Employee.employer.amount_position
-  end
+  private
 
-  def self.find_date_earning
-  	Employee.employer.date_earning_position
+  def find_employee_id
+    self.employee_id = Employee.find_by(external_ref: external_ref).id
```

Comment:
I can remove the exception and change the method to ->  _self.employee_id ||= Employee.find_by(external_ref: external_ref).try(:id)_ in case not found will return nil and the error will be handling on import and also solves your comment about the exception.